src/plans/create/hooks/useInitializeMappings.ts

```diff
@@ -0,0 +1,94 @@
+import { useMemo } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { isEmpty } from '@utils/helpers';
+
+import type { MappingValue } from '../types';
+
+type MappingFieldIds = {
+  sourceField: string;
+  targetField: string;
+  mapField: string;
+};
+
+type UseInitializeMappingsParams<T extends Record<string, unknown>> = {
+  isLoading: boolean;
+  currentMap: T[] | undefined;
+  usedSources: MappingValue[];
+  defaultTarget: MappingValue;
+  fieldIds: MappingFieldIds;
+  defaultTargetName?: string;
+};
+
+/**
+ * Automatically creates mappings for unmapped sources by pairing them with default targets
+ */
+export const useInitializeMappings = <T extends Record<string, unknown>>({
+  currentMap,
+  defaultTarget,
+  defaultTargetName,
+  fieldIds,
+  isLoading,
+  usedSources,
+}: UseInitializeMappingsParams<T>): void => {
+  const { setValue } = useFormContext();
+
+  useMemo(() => {
```

Comment:
why useMemo and not useEffect?

Reply:
I did try a useEffect here, but using useMemo prevents a visual flicker because it calculates the value synchronously during the render cycle, and useEffect would run after the component renders, causing an initial render with stale data followed by a re-render with the updated value.

Reply:
That's fair and a good point. Will update